File: Data Structures/Insertion Sort/InsertionSort.java
package datastructures;

public class InsertionSort {
    public int[] sort(int[] arr){
        for(int i = 0;i < arr.length;i = i + 1){
            int current_val = arr[i];
            int comp_place = i - 1;
            while(comp_place >= 0 & current_val < arr[comp_place]){
                arr[comp_place + 1] = arr[comp_place];
                comp_place = comp_place - 1;
            }
            arr[comp_place + 1] = current_val;
        }
        return arr;
    }
    public static void main(String args[]){
        InsertionSort is = new InsertionSort();
        int arr[] = {4,5,3,2,1};
        int sorted_arr[] = is.sort(arr);
        for(int i = 0;i < 5;i = i + 1){
            System.out.print(sorted_arr[i]);
        } 
    }
}

File: Data Structures/Insertion Sort/InsertionSort.py
class InsertionSort():
    def __init__(self):
        None
    def sort(self,arr):
        for i in range(1,len(arr)):
            current_val = arr[i]
            comp_place = i - 1
            while comp_place >= 0 and current_val < arr[comp_place]:
                arr[comp_place + 1] = arr[comp_place]
                comp_place = comp_place - 1
            arr[comp_place + 1] = current_val
        return arr

Inssort = InsertionSort()
arr = [1,2,3,8,4,5,11,6]
Inssort.sort(arr)
print(arr)
        
